src/main/kotlin/compiler/Coder.kt

```diff
@@ -54,10 +54,11 @@ class Coder(val ast: Node) {
     // Nodes call this to jump to a named future address.
     fun jumpForward(from: Node, name: String) {
         val address = mem.size
-        if (forwardJumps.containsKey(name)) {
-            forwardJumps[name]!!.add(address)
+        val fullname = "$name${from.id}"
```

Comment:
I moved adding the Node's unique ID to the jump tag here in Coder, to simplify.  Most of these Coder changes are that refactor.